lib/mutant/parser.rb

```diff
@@ -12,18 +12,18 @@ def initialize
       @cache = {}
     end
 
-    # Root node parsed from file
+    # Parse path into AST
     #
-    # @param [#to_s] path
+    # @param [Pathname] path
     #
     # @return [AST::Node]
     #
     # @api private
-    def parse(path)
+    def call(path)
       @cache.fetch(path) do
-        @cache[path] = Parser::CurrentRuby.parse(File.read(path))
+        @cache[path] = ::Parser::CurrentRuby.parse(path.read)
```

Comment:
Yeah, I think the elimination of redundant portion of `@cache[k]` and `@cache[k]=` weights more than my argument above. Changing to your suggestion.